File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/config/Condition.java
package com.quicksilver.objectvalidator.config;

public class Condition {
    public final String type, field, arg;
    public final String[] args;
    public final Condition[] conditions;

    public Condition(String type, String field, String arg, String[] args, Condition[] conditions) {
        this.type = type;
        this.field = field;
        this.arg = arg;
        this.args = args;
        this.conditions = conditions;
    }

    @SuppressWarnings("unused")
    private Condition() {
        this(null, null, null, null, null);
    }
}

File: C Sharp/Object Validator/Config/Condition.cs
namespace Quicksilver.ObjectValidator.Config;
public class Condition(string type, string? field, string? arg, string?[]? args, Condition[]? conditions)
{
    public readonly string type = type;
    public readonly string? field = field, arg = arg;
    public readonly string?[]? args = args;
    public readonly Condition[]? conditions = conditions;
}